libpod/runtime_pod.go

```diff
@@ -90,18 +90,16 @@ func (r *Runtime) LookupPod(idOrName string) (*Pod, error) {
 // output. Multiple filters are handled by ANDing their output, so only pods
 // matching all filters are returned
 func (r *Runtime) Pods(filters ...PodFilter) ([]*Pod, error) {
-	r.lock.RLock()
-	defer r.lock.RUnlock()
-
-	if !r.valid {
-		return nil, define.ErrRuntimeStopped
-	}
-
-	pods, err := r.state.AllPods()
+	pods, err := r.GetAllPods()
 	if err != nil {
 		return nil, err
 	}
 
+	return r.FilterPodsWithFilterFunc(pods, filters...)
```

Comment:
Why is this changed? I don't see any uses of FilterPodsWithFilterFunc. We also dropped the runtime validity check and lock?

Reply:
Fixed now. I added this to support filters on the latest pod but this is no longer needed.  No, we did not drop that check there or was already a method [GetAllPods](https://github.com/containers/libpod/blob/master/libpod/runtime_pod.go#L121) that does the same just using that.